obstacle_avoidance_VL53L0X.ino

```diff
@@ -0,0 +1,51 @@
+/* This example shows how to use continuous mode to take
+range measurements with the VL53L0X. It is based on
+vl53l0x_ContinuousRanging_Example.c from the VL53L0X API.
+
+The range readings are in units of mm. */
+
+#include <Wire.h>
+#include <VL53L0X.h>
+#include <Smartcar.h>
+#include <Wire.h>
+#include <VL53L0X.h>
+
+VL53L0X sensor;
+BrushedMotor leftMotor(smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
+
+void setup()
+{
+  Serial.begin(9600);
+  Wire.begin();
+
+  sensor.setTimeout(500);
+  if (!sensor.init())
+  {
+    Serial.println("Failed to detect and initialize sensor!");
+    while (1) {}
+  }
+
+  // Start continuous back-to-back mode (take readings as
+  // fast as possible).  To use continuous timed mode
+  // instead, provide a desired inter-measurement period in
+  // ms (e.g. sensor.startContinuous(100)).
+  sensor.startContinuous();
+ 
+}
+
+void loop()
+{
+  Serial.print(sensor.readRangeContinuousMillimeters());
+  if (sensor.timeoutOccurred()) { Serial.print(" TIMEOUT"); }
+  Serial.println();
+   car.setSpeed(50);
+  while(sensor.readRangeContinuousMillimeters()!=0 && sensor.readRangeContinuousMillimeters()<200)
```

Comment:
You are making two separate measurements. In other words you may not necessarily read the same data both times. The first one could be 210 and the other one 199, which will induce a wrong behavior.